src/arrayMethodJoin.js

```diff
@@ -4,8 +4,30 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let correctSeparator = separator;
+
+    if (separator === null) {
+      correctSeparator = 'null';
+    }
+
+    let newString = '';
+
+    for (let i = 0; i < this.length; i++) {
+      let element = this[i];
+
+      if (element === undefined || element === null) {
+        element = '';
+      }
+
+      if (i === this.length - 1) {
+        newString += element;
+      } else {
+        newString += element + correctSeparator;
+      }
```

Comment:
not critical, but this part can be simplified
hint: you can always add elements if they exist, and only add a delimiter if it's not the last element